qutebrowser/config/config.py

```diff
@@ -657,6 +672,50 @@ def set_command(self, win_id, section_=None, option=None, value=None,
             message.info(win_id, "{} {} = {}".format(
                 section_, option, val), immediately=True)
 
+    @cmdutils.register(instance='config')
+    def reload_config(self):
+        """Reload the config & keyconfig files."""
+        old_config = objreg.get('config')
+        objreg.delete('config')
+        try:
+            _init_main_config(QApplication.instance())
+        # Restore the old config if we failed
+        except:
+            if objreg.get('config', None) is not None:
+                objreg.delete('config')
+            objreg.register('config', old_config)
+            return
+
+        # Run _after_set() to notify code the setting has changed, but only for
+        # settings that actually changed value
+        for sect, settings in objreg.get('config').sections.items():
+            for key, _value in settings.items():
+                try:
+                    old_value = old_config.get(sect, key)
+                except:
+                    continue
+
+                new_value = objreg.get('config').get(sect, key)
+                if old_value == new_value:
+                    continue
+                log.config.debug('Setting {}.{} changed from {} to {}'.format(
+                                 sect, key, old_value, new_value))
+                self._after_set(sect, key)
+
+        # Also reload the key config
+        # TODO: Doesn't seem to take effect
+        old_key_config = objreg.get('key-config')
+        objreg.delete('key-config')
+        try:
+            _init_key_config(QApplication.instance())
+        except:
```

Comment:
Maybe it doesn't do anything because it raises an exception and you're hiding it here? :wink: